src/controller_examples/zookeeper_controller/exec/zookeeper_lib/helper_funcs.rs

```diff
@@ -9,62 +9,65 @@ use vstd::{prelude::*, string::*};
 
 verus! {
 
+pub struct ZKNodeResult {
+    pub res: Result<(), Error>,
+}
+
 struct LoggingWatcher;
 impl Watcher for LoggingWatcher {
     #[verifier(external)]
     fn handle(&self, e: WatchedEvent) {}
 }
 
 #[verifier(external)]
-pub fn reconcile_zk_node(zk: &ZookeeperCluster) -> Result<(), Error>
-    requires
-        zk@.metadata.name.is_Some(),
-        zk@.metadata.namespace.is_Some(),
-{
-    let path = cluster_size_zk_node_path(&zk);
+pub fn reconcile_zk_node(path: String, uri: String, replicas: String) -> ZKNodeResult {
     let zk_client = ZooKeeper::connect(
-        zk_service_uri(&zk).as_str().into_rust_str(),
+        uri.as_str().into_rust_str(),
         Duration::from_secs(10),
         LoggingWatcher,
     );
     if zk_client.is_err() {
-        return Result::Err(Error::ClusterSizeZKNodeCreationFailed);
+        return ZKNodeResult{ res: Err(Error::ClusterSizeZKNodeCreationFailed) };
     }
     let zk_client = zk_client.unwrap();
     let path_look_up = zk_client.exists(path.as_rust_string_ref(), false);
     if path_look_up.is_err() {
-        return Result::Err(Error::ClusterSizeZKNodeCreationFailed);
+        return ZKNodeResult{ res: Err(Error::ClusterSizeZKNodeCreationFailed) };
     }
     let path_look_up = path_look_up.unwrap();
     match path_look_up {
-        Some(_) => Ok(()),
+        Some(_) => {
+            return ZKNodeResult{ res: Ok(())};
+        },
         None => {
             // First create the parent node
             if zk_client
                 .create(
                     "/zookeeper-operator",
-                    new_strlit("CLUSTER_SIZE=").to_string().concat(i32_to_string(zk.spec().replicas()).as_str())
+                    new_strlit("CLUSTER_SIZE=").to_string().concat(replicas.as_str())
```

Comment:
Why do we need to write the CLUSTER_SIZE to the parent node? I thought that only the value in the child node matters (to the teardown script). Please correct me if I am wrong @ZichengMa @euclidgame

Reply:
Oh I'm idiot on this, all I do is to follow the reference controller. Can you check the reference controller?

Reply:
Confirmed that we don't need to set the value to the parent node.